ruuvi_examples/ruuvi_firmware/main.c

```diff
@@ -282,97 +296,149 @@ ret_code_t lis2dh12_int2_handler(const ruuvi_standard_message_t message)
+ All subsystems are initalized and any failures are noted and available later in init_status
+ Since some events occur after tag is deployed and no one can see the LEDs the system continues operating.
+
+ After initalizition (including setting up interrupts)
+    we loop here calling app_sched_execute and sd_app_evt_wait 
+*/
 int main(void)
 {
-  ret_code_t err_code = NRF_SUCCESS; // Error flag, OR any errors to it and check the error code at the end.
+ init_leds();                      // LEDs first (they're easy and cannot fail)  drivers/init/init.c ( RED got set up earlier)
+ nrf_delay_ms(1000u);              // sit here for a second with LEDs off
+
+  // Announce version
+ for ( int16_t i=0; i<3; i++){   // 3 segment version
+    for ( int16_t v=0; v < version[i]; v++){RED_LED_ON;nrf_delay_ms(100u);RED_LED_OFF; nrf_delay_ms(500u); }
+    nrf_delay_ms(1000u);  }
+  
+
+  // start watchdog now incase some initalization hangs up
+//  init_watchdog(NULL);  // watchdog_default_handler ONLY LOG_ERROR and returns to ??? 
+               // do something exciting , wait a few seconds and then keep going (i.e. never permit RESET!)       ToDo (_)
 
-  // Initialize log.
-  err_code |= init_log();
+//for ( int16_t i=0; i<13; i++){ RED_LED_ON; nrf_delay_ms(100u); RED_LED_OFF; nrf_delay_ms(200u); } // we got this far DEBUGING
 
-  // Setup leds. LEDs are active low, so setting high them turns leds off.
-  err_code |= init_leds();      // INIT leds first and turn RED on.
-  nrf_gpio_pin_clear(LED_RED);  // If INIT fails at later stage, RED will stay lit.
+  if( init_log() )                                                              init_status |=LOG_FAILED_INIT;       
+   else NRF_LOG_DEBUG("LOG initalized \r\n"); // subsequent initalizations assume log is working
 
-  //Init NFC ASAP in case we're waking from deep sleep via NFC (todo)
-  set_nfc_callback(app_nfc_callback);
-  err_code |= init_nfc();
 
-  if (NRF_SUCCESS == init_sensors()) { model_plus = true; }
+  battery_voltage_init();  // function with no return code (NEEDS rewrite)
+   uint16_t vbat= getBattery();
+   if(      vbat < BATTERY_MIN_V )                                             init_status |=BATTERY_FAILED_INIT;  
+    else NRF_LOG_DEBUG("BATTERY initalized \r\n"); 
 
-  // Initialize BLE Stack. Required in all applications for timer operation.
-  err_code |= init_ble();
+ if(init_sensors() == NRF_SUCCESS )  {   model_plus = true; // really need to be one at a time!!
+    NRF_LOG_DEBUG("Sensors initalized \r\n");  }
+
+  set_nfc_callback(app_nfc_callback);   // can this fail?
+//Init NFC ASAP in case we're waking from deep sleep via NFC (todo)
+ if( init_nfc() )                                                               init_status |= NFC_FAILED_INIT ; 
+  else NRF_LOG_DEBUG("NFC init \r\n");
+
+// Start (?) interrupts.  can this fail?
+ if( pin_interrupt_init() )                                                     init_status |= PIN_INT_FAILED_INIT ; 

```

Comment:
This is also internal to nRF52, it should not fail and if it does the code itself could not detect the failure if I recall correctly